packages/api-queries/src/me-preferences.ts

```diff
@@ -46,8 +46,19 @@ export const userPreferenceMutation = < P extends keyof UserPreferences >( prefe
 				[ preferenceName ]: data,
 			} as Partial< UserPreferences > ),
 		onSuccess: ( newData ) => {
-			queryClient.setQueryData( rawUserPreferencesQuery().queryKey, ( oldData ) =>
-				oldData ? { ...oldData, ...newData } : newData
-			);
+			queryClient.setQueryData( rawUserPreferencesQuery().queryKey, ( oldData ) => {
+				if ( ! oldData ) {
+					return newData;
+				}
+
+				// If newData doesn't contain the preference key, it means it was unset.
+				// We need to remove it from oldData.
+				if ( ! ( preferenceName in newData ) ) {
+					const { [ preferenceName ]: _, ...rest } = oldData;
+					return rest;
```

Comment:
I'm not sure but should it be `{ ...rest, ...newData }` 🤔

Reply:
Hmm good catch, `{ ...rest, ...newData }` is technically more robust as it also includes new values. But I'm not sure if that would happen since we're just updating a single preference. In any case I'll update it, better safe than sorry!